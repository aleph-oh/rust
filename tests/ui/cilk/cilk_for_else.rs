#![feature(cilk)]
// Tests a simple cilk_for loop with an else arm to show that it errors.

// compile-flags: -C panic=abort
// no-prefer-dynamic

use std::sync::atomic::AtomicU64;

fn main() {
    let sum: AtomicU64 = AtomicU64::new(0);
    cilk_for i in 1..10 {
        sum.fetch_add(i, std::sync::atomic::Ordering::Relaxed);
    } else { //~ ERROR: `for...else` loops are not supported
        panic!("shouldn't get here!");
    }
    assert_eq!(sum.into_inner(), 45);
}
