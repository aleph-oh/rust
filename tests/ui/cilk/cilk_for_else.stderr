error: `for...else` loops are not supported
  --> $DIR/cilk_for_else.rs:13:7
   |
LL |       cilk_for i in 1..10 {
   |       -------- `else` is attached to this loop
LL |           sum.fetch_add(i, std::sync::atomic::Ordering::Relaxed);
LL |       } else {
   |  _______^
LL | |         panic!("shouldn't get here!");
LL | |     }
   | |_____^
   |
   = note: consider moving this `else` clause to a separate `if` statement and use a `bool` variable to control if it should run

error: aborting due to 1 previous error

